cgi-bin/DW/Controller/Shop/Account.pm

```diff
@@ -0,0 +1,283 @@
+            # conflict or something
+
+            my $item = DW::Shop::Item::Account->new(
+                type           => $post->{accttype},
+                user_confirmed => $post->{alreadyposted},
+                force_spelling => $post->{force_spelling},
+                %item_data
+            );
+
+            # check for renewing premium as paid
+            my $u           = LJ::load_userid( $item ? $item->t_userid : undef );
+            my $paid_status = $u ? DW::Pay::get_paid_status($u) : undef;
+
+            if ($paid_status) {
+                my $paid_curtype = DW::Pay::type_shortname( $paid_status->{typeid} );
+                my $has_premium  = $paid_curtype eq 'premium' ? 1 : 0;
+
+                my $ok = DW::Shop::Item::Account->allow_account_conversion( $u, $item->class );
+
+                if ( $ok && $has_premium && $item->class eq 'paid' && !$post->{prem_convert} ) {
+
+                    # check account expiration date
+                    my $exptime = DateTime->from_epoch( epoch => $paid_status->{expiretime} );
+                    my $newtime = DateTime->now;
+
+                    if ( my $future_ymd = $item->deliverydate ) {
+                        my ( $y, $m, $d ) = split /-/, $future_ymd;
+                        $newtime = DateTime->new( year => $y + 0, month => $m + 0, day => $d + 0 );
+                    }
+
+                    my $to_day = sub { return $_[0]->truncate( to => 'day' ) };
+
+                    if ( DateTime->compare( $to_day->($exptime), $to_day->($newtime) ) ) {
+                        my $months = $item->months;
+                        my $newexp = $exptime->clone->add( months => $months );
+                        my $paid_d = $exptime->delta_days($newexp)->in_units('days');
+
+                        # FIXME: this should be DW::BusinessRules::Pay::DWS::CONVERSION_RATE
+                        my $prem_d = int( $paid_d * 0.7 );
+
+                        my $ml_args =
+                            { date => $exptime->ymd, premium_num => $prem_d, paid_num => $paid_d };
+
+                        # but only include date if the logged-in user owns the account
+                        delete $ml_args->{date} unless $remote && $remote->has_same_email_as($u);
+
+                        $errors->add( undef, '.error.premiumconvert',          $ml_args );
+                        $errors->add( undef, '.error.premiumconvert.postdate', $ml_args )
+                            if $ml_args->{date};
+                        $vars->{premium_convert} = 1;
+
+                    }
+                }
+            }
+
+            unless ( $errors->exist ) {
+
+                my ( $rv, $err ) = $rv->{cart}->add_item($item);
+                $errors->add( '', $err ) unless $rv;
+
+                unless ( $errors->exist ) {
+                    return $r->redirect("$LJ::SITEROOT/shop");
+                }
+            }
+        }
+
+    }
+
+    $vars->{errors} = $errors;
+
+    sub get_opts {
```

Comment:
If you want a local sub, you should do `my $get_opts = sub...` here; otherwise move it out to the top level of the module.